activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -141,6 +141,14 @@ def supports_datetime_with_precision?
         end
       end
 
+      def supports_virtual_columns?
+        if mariadb?
+          version >= "5.2.0"
+        else
+          version >= "5.7.5"
```

Comment:
I implemented extracting the expression for MariaDB in e0841ac6aaa873a7125342d4465c28e00dd919e3. Thanks!